script.js

```diff
@@ -1,4 +1,398 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+var inputScissors = "scissors";
+var inputPaper = "paper";
+var inputStone = "stone";
+var inputReversedScissors = "reversed scissors";
+var inputReversedPaper = "reversed paper";
+var inputReversedStone = "reversed stone";
+// modes of games
+var currentGameMode = "user to input username";
+var userName = " ";
+var playerOrComputerMode = " ";
+var normalOrKoreanVersion = "";
+// score board
+var draw = 0;
+var playerWinCount = 0;
+var programWinCount = 0;
+var computerWinCount = 0;
+var totalCount = 0;
+// korean version
+var mukJjiPpaState = "play again";
+var tempWinner = " ";
+
+var randomNumber1to3 = function () {
+  var randomDecimal = Math.random() * 3; // >= 0 to < 3
+  var randomInteger = Math.floor(randomDecimal); // 0,1,2
+  var randomNumber = randomInteger + 1; // 1,2,3
+  return randomNumber;
+};
+
+// computer generates "scissors" "paper" "stone"
+var outputByComputer = function () {
+  var randomNumber = randomNumber1to3();
+  var programOutput = "scissors";
+  if (randomNumber == "1") {
+    var programOutput = "paper";
+  }
+  if (randomNumber == "2") {
+    var programOutput = "stone";
+  }
+  return programOutput;
+};
+
+// random number to be generated for "reversed" function
+var randomNumber1to6 = function () {
+  var randomDecimal = Math.random() * 6; // >= 0 to < 6
+  var randomInteger = Math.floor(randomDecimal); // 0,1,2,3,4,5
+  var randomNumber = randomInteger + 1; // 1,2,3,4,5,6
+  return randomNumber;
+};
+
+// program generates "scissors" "paper" "stone" "reversed scissors" "reversed paper" "reversed stone"
+var outputByProgram = function () {
+  // computer VS computer mode
+  var randomNumber = randomNumber1to6();
+  var programOutput = "scissors";
+  if (randomNumber == "1") {
+    var programOutput = "paper";
+  }
+  if (randomNumber == "2") {
+    var programOutput = "stone";
+  }
+  if (randomNumber == "3") {
+    var programOutput = "reversed scissors";
+  }
+  if (randomNumber == "4") {
+    var programOutput = "reversed paper";
+  }
+  if (randomNumber == "5") {
+    var programOutput = "reversed stone";
+  }
+  return programOutput;
+};
+
+var winLoseBoard = function (playerToComputer) {
+  if (
+    playerToComputer == ["scissors" + "scissors"] ||
+    playerToComputer == ["paper" + "paper"] ||
+    playerToComputer == ["stone" + "stone"] ||
+    playerToComputer == ["reversed scissors" + "scissors"] ||
+    playerToComputer == ["reversed paper" + "paper"] ||
+    playerToComputer == ["reversed stone" + "stone"]
+  ) {
+    draw = draw + 1;
+    var myOutputValue = `It's a draw!<br><br>Player Win Count: ${playerWinCount}.<br>Player Lose Count: ${computerWinCount}.<br>Draw Count: ${draw}.`;
+  }
```

Comment:
myOutputValue variable should be created outside of the if statement. This is because if the first if statement does not run, myOutputValue will not be created for subsequent references to myOutputValue. The reason why this does not throw you an error now is because JS will automatically know to create a variable if the variable name does not exist. However, it would be good code practice to indicate new variables with 'var <variablename>' in your code.